src/Persistence/PersistentObjectFactory.php

```diff
@@ -454,6 +456,10 @@ final protected function initializeInternal(): static
         return parent::initializeInternal()
             ->afterInstantiate(
                 static function(object $object, array $parameters, PersistentObjectFactory $factoryUsed): void {
+                    if (PHP_VERSION_ID >= 80400) {
```

Comment:
We need to make sure we don't track objects creates with `PersistentProxyObjectFactory`